python_on_whales/components/container/cli_wrapper.py

```diff
@@ -1829,14 +1818,15 @@ def unpause(self, x: Union[ValidContainer, List[ValidContainer]], /):
             x: One or more containers (name, id or `python_on_whales.Container` object).
 
         # Raises
-            `python_on_whales.exceptions.NoSuchContainer` if the container does not exists.
+            `python_on_whales.exceptions.NoSuchContainer` if any of the
+            containers do not exist.
         """
-        x = to_list(x)
-        if x == []:
+        containers = to_list(x)
+        if len(containers) == 0:
             # nothing to do
             return
         full_cmd = self.docker_cmd + ["container", "unpause"]
-        full_cmd += x
+        full_cmd.extend(str(c) for c in containers)
```

Comment:
To avoid forgetting the conversion to `str` every time we add something to `full_cmd`, maybe we should make `full_cmd` being a `list[Stringable]` with Stringable being a typing.Protocol that ensures that a method `__str__` is present. Then we can do `str()` when we call the function `run()`. What do you think? That would remove some complexity in the implementation of commands.

Reply:
Yeah, after making some of these changes I realised that `run()` already handles converting all args to strings, so this isn't strictly necessary... One option would be to reflect that in the type signature, although it does have the slight downside of accidentally passing in an object that doesn't have a suitable `__str__()` and it not being obvious when it doesn't work properly. Note that all objects are "stringable" in python.

Reply:
Yeah haha, but it's the same issue here by calling `str()` manually, we can make the mistake and call it on the wrong objects.

Reply:
Well, I guess it's a bit more explicit when done here than in `run()`. But changing `run()` to *not* do it would presumably be difficult at this point. I started making these changes based on it being done in some methods and not others, before I realised `run()` does it automatically. Are you saying you want me to revert these stringify changes?

Reply:
`run()` already does convert to string, so for me the changes you made are not needed. You can revert back to how it was.